src/features/projects/components/maps/PlantLocations.tsx

```diff
@@ -0,0 +1,236 @@
+import React, { ReactElement } from 'react';
+import { Layer, Marker } from 'react-map-gl';
+import { Source } from 'react-map-gl';
+import { ProjectPropsContext } from '../../../common/Layout/ProjectPropsContext';
+import { useRouter } from 'next/router';
+import styles from '../../styles/PlantLocation.module.scss';
+import * as turf from '@turf/turf';
+import { localizedAbbreviatedNumber } from '../../../../utils/getFormattedNumber';
+import i18next from '../../../../../i18n';
+
+const { useTranslation } = i18next;
+
+interface Props {}
+
+export default function PlantLocations({}: Props): ReactElement {
+  const router = useRouter();
+  const {
+    plantLocations,
+    hoveredPl,
+    selectedPl,
+    setSelectedPl,
+    setHoveredPl,
+    viewport,
+  } = React.useContext(ProjectPropsContext);
+
+  const { i18n, t } = useTranslation(['common']);
+
+  const openPl = (pl: any) => {
+    setSelectedPl(pl);
+  };
+
+  const onHover = (pl: Object) => {
+    setHoveredPl(pl);
+  };
+
+  const onHoverEnd = (pl: Object) => {
+    if (
+      hoveredPl &&
+      (hoveredPl.type === 'single' || hoveredPl.type === 'sample')
+    )
+      setHoveredPl(null);
+  };
+
+  const getPlTreeCount = (pl: any) => {
+    let count = 0;
+    if (pl && pl.plantedSpecies) {
+      for (const key in pl.plantedSpecies) {
+        if (Object.prototype.hasOwnProperty.call(pl.plantedSpecies, key)) {
+          const element = pl.plantedSpecies[key];
+          count += element.treeCount;
+        }
+      }
+      return count;
+    } else {
+      return 0;
+    }
+  };
+
+  const getPlArea = (pl: any) => {
+    if (pl && pl.type === 'multi') {
+      const area = turf.area(pl.geometry);
+      return area / 10000;
+    } else {
+      return 0;
+    }
+  };
+
+  const getPolygonColor = (pl: any) => {
+    const treeCount = getPlTreeCount(pl);
+    const plantationArea = getPlArea(pl);
+    const density = treeCount / plantationArea;
+    if (density > 2500) {
+      return 0.5;
+    } else if (density > 2000) {
+      return 0.4;
+    } else if (density > 1600) {
+      return 0.3;
+    } else if (density > 1000) {
+      return 0.2;
+    } else {
+      return 0.1;
+    }
+  };
+
+  const getDateDiff = (pl: any) => {
+    const today = new Date();
+    const plantationDate = new Date(pl.plantDate);
+    const differenceInTime = today.getTime() - plantationDate.getTime();
+    const differenceInDays = differenceInTime / (1000 * 3600 * 24);
+    if (differenceInDays < 1) {
+      return t('today');
+    } else if (differenceInDays < 2) {
+      return t('yesterday');
+    } else if (differenceInDays < 30) {
+      return `${localizedAbbreviatedNumber(
+        i18n.language,
+        differenceInDays,
+        0
+      )}${t('d')} ${t('ago')}`;
```

Comment:
This should best be only one resource with a parameter: e.g.

```JavaScript
t('days_ago', {
  days: localizedAbbreviatedNumber( ​i18n.language, ​differenceInDays, ​0)
})
```

Reply:
This makes it easier to translate it. Otherwise we might get in trouble with some languages.